### PR TITLE
Email + SMS verification code

### PR DESCRIPTION
**Pull request by AlekseiPetrovDev**
https://user-images.githubusercontent.com/78972087/164085232-6ab8bda6-0b3d-4412-9745-add7889ced7b.mov

Email verification logic

### CODE REVIEW

File: packages/dialect-react-ui/components/Icon/Resend.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import * as React from 'react';
+import { SVGProps } from 'react';
+
+const SvgResend = (props: SVGProps<SVGSVGElement>) => (
+  <svg
+    width={14}
+    height={17}
+    fill="none"
```
Comment (fsher) [inline packages/dialect-react-ui/components/Icon/Resend.tsx:8]: Please add a viewBox prop here as well.

File: packages/dialect-react/components/ApiContext/index.tsx [hunk 2]
```diff
@@ -201,6 +210,42 @@ export const ApiProvider = ({ dapp, children }: PropsType): JSX.Element => {
...
+      if (!isWalletConnected || !dapp) return;
+      setSendingCode(true);
+      try {
+        await resendCode(wallet, dapp, address);
+        setSendingCode(false);
+        setVerificationCodeError(null)
+      } catch (err) {
+        setVerificationCodeError(err as ParsedErrorData)
+        throw err;
+      }finally {
```
Comment (fsher) [inline packages/dialect-react/components/ApiContext/index.tsx:243]: Seems like formatting is a bit wrong here... Try running prettier :)

**Review by fsher — APPROVED** [review]
Great job! Please fix formatting, looks good to me otherwise.

**Review by cbosborn — APPROVED** [review]
lgtm 🎉

**Pull request merged**